crates/matrix-sdk/src/event_handler/maps.rs

```diff
@@ -197,3 +209,17 @@ impl<'a> Borrow<KindTypeRoomId<'a>> for KindTypeRoomIdWrap {
         &self.0
     }
 }
+
+// Arcane Rust magic 🧙
+pub(crate) fn try_downcast<T, K>(k: K) -> Result<T, K>
```

Comment:
This exists because downcast methods return either a ref or a mut ref, right?

Reply:
Yeah, there's no non-reference casting in `std`. Could have pulled in the `castaway` crate for this as an alternative, but that actually includes a bunch of extra complexity for non-`'static` types IIRC.